spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -519,6 +523,253 @@ public void testUpdateRefreshesRelationCache() {
     spark.sql("UNCACHE TABLE tmp");
   }
 
+  @Test
+  public void testUpdateWithInSubquery() {
```

Comment:
I validated results match Postgres for some queries. I'll check others before the PR is merged.

Reply:
One thing I think you may want to check here is subqueries that uses aliases as well

Reply:
Good point, let me add this together with self joins.

Reply:
Added `testUpdateWithSelfSubquery` with this.